### PR TITLE
Add v1 controller/APIs and support multi-version

### PR DESCRIPTION
**Pull request by terrytangyuan**
Fixes https://github.com/kubeflow/xgboost-operator/issues/72.

/cc @merlintang @Jeffwan 

Signed-off-by: terrytangyuan <terrytangyuan@gmail.com>

### CODE REVIEW

File: cmd/manager/main.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ import (
 	"os"
 
 	"github.com/kubeflow/xgboost-operator/pkg/apis"
-	"github.com/kubeflow/xgboost-operator/pkg/controller"
+	controller "github.com/kubeflow/xgboost-operator/pkg/controller/v1"
```
Comment (Jeffwan) [inline cmd/manager/main.go:23]: This means the master will point to v1 changes. This is expected, right?

**Comment by Jeffwan** [comment]
Can you check if Makefile is still working?  It pass CI and I think there's no more changes needed on ./travis.yaml

File: cmd/manager/main.go (hunk 1 above)
Comment (terrytangyuan) [reply cmd/manager/main.go:23]: Yes. This is intended so more users would start testing v1 by default.

**Comment by terrytangyuan** [comment]
@Jeffwan Yes, I think Makefile still needs some updates. For example, `make manifests` still uses `vendor` directory that we no longer have. We need to start testing/using Makefile on Travis.

**Pull request merged**